advanced/core.py

```diff
@@ -3,20 +3,31 @@
 import python_storage_balancing
 import session_vars
 
+from genquery import *
+
 #def pep_resource_resolve_hierarchy_pre(rule_args, callback, rei):
 #    return python_storage_balancing.pep_resource_resolve_hierarchy_pre(rule_args, callback, rei)
 
-def acPostProcForPut(rule_args, callback, rei):
-    sv = session_vars.get_map(rei)
-    phypath = sv['data_object']['file_path']
-    objpath = sv['data_object']['object_path']
+def pep_api_data_obj_put_post(rule_args, callback, rei):
+    import os
+
+    data_obj_inp = rule_args[2]
+    obj_path = str(data_obj_inp.objPath)
+    resc_hier = str(data_obj_inp.condInput['resc_hier'])
+    query_condition_string = f'COLL_NAME = \'{os.path.dirname(obj_path)}\' and '  \
+                             f'DATA_NAME = \'{os.path.basename(obj_path)}\' and ' \
+                             f'DATA_RESC_HIER = \'{resc_hier}\''
+
+    phypath = list(Query(callback, 'DATA_PATH', query_condition_string))[0]
```

Comment:
Not shown here: A real policy should consider the host executing this rule and compare with the hostname of the resource. Figured this can be addressed if the room is feeling it

Reply:
or capture that 'production concern' in a comment

that's enough for training - gets people thinking - and makes it more likely someone doesn't just copy paste this into their real stuff.